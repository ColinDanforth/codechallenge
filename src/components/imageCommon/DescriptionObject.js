import React from 'react'
import PropTypes from 'prop-types'

const descriptionStyle={
  width: '90%',
  height: 'auto',
  margin: '3%',
  flex: "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
}

const DescriptionObject = (props) => {
  if(props.description !== undefined && props.description !== null && props.description.length > 0) {
    return <div style={descriptionStyle} dangerouslySetInnerHTML={{__html: props.description}}/>
  }else{
    return <div style={{'display': 'none'}}/>
  }
}

export default DescriptionObject

DescriptionObject.propTypes={
  description: PropTypes.string
}
