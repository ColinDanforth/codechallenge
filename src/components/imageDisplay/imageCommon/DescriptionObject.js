import React from 'react'
import PropTypes from 'prop-types'

const descriptionStyle={
  minWidth: '600px',
  maxWidth: '820px',
  width: 'auto',
  height: 'auto',
  margin: '3%',
}

const DescriptionObject = (props) => {
  if(props.description !== undefined && props.description !== null && props.description.length > 0) {
    return <div style={descriptionStyle} dangerouslySetInnerHTML={{__html: props.description}}/>
  }else{
    return <div style={{'display': 'none'}}/>
  }
}

export default DescriptionObject

DescriptionObject.propTypes={
  description: PropTypes.string
}
