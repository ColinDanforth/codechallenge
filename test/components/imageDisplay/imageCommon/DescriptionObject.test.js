import React from 'react';
import DescriptionObject from '../../../../src/components/imageCommon/DescriptionObject';
import renderer from 'react-test-renderer';

test('Description should display test string', () => {
  const component = renderer.create(
    <DescriptionObject description="Test String"/>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

})

test(`Description should display test string with <a> element`, () => {
  const component = renderer.create(
    <DescriptionObject description="Test String <a src='http://google.com'/> "/>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

})

test('Description should display as a div with a style of display none', () => {
  const component = renderer.create(
    <DescriptionObject description=""/>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

})
