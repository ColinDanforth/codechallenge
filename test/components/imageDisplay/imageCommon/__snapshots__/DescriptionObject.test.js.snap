// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description should display as a div with a style of display none 1`] = `
<div
  style={
    Object {
      "display": "none",
    }
  }
/>
`;

exports[`Description should display test string 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "Test String",
    }
  }
  style={
    Object {
      "height": "auto",
      "margin": "3%",
      "maxWidth": "820px",
      "minWidth": "600px",
      "width": "auto",
    }
  }
/>
`;

exports[`Description should display test string with <a> element 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "Test String <a src='http://google.com'/> ",
    }
  }
  style={
    Object {
      "height": "auto",
      "margin": "3%",
      "maxWidth": "820px",
      "minWidth": "600px",
      "width": "auto",
    }
  }
/>
`;
